models/modules/resnet_architecture/resnet_generator.py

```diff
@@ -274,19 +282,14 @@ def forward(self, input, extract_layer_ids=[], encode_only=False,get_attention_m
         
         if -1 in extract_layer_ids: #if -1 is in extract_layer_ids, the output of the encoder will be returned (features just after the last layer) 
             extract_layer_ids.append(len(self.resnet_blocks))
-        if len(extract_layer_ids) > 0:
-            feat=x
-            feats=[]
-            for layer_id, layer in enumerate(self.resnet_blocks):
-                feat = layer(feat)
-                if layer_id in extract_layer_ids:
-                    feats.append(feat)
-            if encode_only:
-                return feats
-            else:
-                x=feat
-        else:
-            x = self.resnet_blocks(x)
+        feat=x
+        feats=[]
+        for layer_id, layer in enumerate(self.resnet_blocks):
+            feat = layer(feat)
+            if layer_id in extract_layer_ids:
+                feats.append(feat)
+                
+        x = self.resnet_blocks(feat)
```

Comment:
Don't compute if not needed (previously "encode_only")? -> split the function in 2